### ISSUE TITLE
Feature Request: Providing Gradients for `layer.set_weights()`

### ISSUE THREAD

**Issue by NuoWenLei** [keras-team/keras#15836]
If you open a GitHub issue, here is our policy:

It must be a bug, a feature request, or a significant problem with the documentation (for small docs fixes please send a PR instead).
The form below must be filled out.

**Here's why we have that policy:**.

Keras developers respond to issues. We want to focus on work that benefits the whole community, e.g., fixing bugs and adding features. Support only helps individuals. GitHub also notifies thousands of people when issues are filed. We want them to see you communicating an interesting problem, rather than being redirected to Stack Overflow.

**System information**
- TensorFlow version (you are using): 2.7.0
- Are you willing to contribute it (Yes/No): No, I am not experienced enough.

**Describe the feature and the current behavior/state.**

Hello, this is also posted in an issue on the tensorflow repo since it does kind of involve gradienttape.

[tensorflow issue link](https://github.com/tensorflow/tensorflow/issues/53541)

The feature is to make `set_weights()` differentiable. The current behavior is that the `model.set_weights()` function breaks the gradient. Given a model named `learner` and a batch of data `batch`

```
l = tf.constant(0.1)
with tf.GradientTape() as gt:
  gt.watch(l)
  new_weights = [w * l for w in learner.weights]

  learner.set_weights(new_weights)

  y_pred = learner(batch["train"]["X"])
  loss = tf.keras.losses.categorical_crossentropy(batch["train"]["y"], y_pred)
grads = gt.gradient(loss, l)
grads # returns None
```

This example returns None for the gradients. In fact, the gradient tape can't calculate the gradient of any variable related to calculating the `new_weights`.

So I was wondering if it's possible to provide gradients for `set_weights()` or if there are any work-arounds to this problem.

**Will this change the current api? How?**

I don't see how this feature would change the current api. It is more a modification on an existing function.

**Who will benefit with this feature?**

I believe adding this feature would make projects related to meta-learning easier to realize.

In my specific case, I was trying to implement [this paper](https://openreview.net/pdf?id=rJY0-Kcll) and translate the [pytorch implementation](https://github.com/markdtw/meta-learning-lstm-pytorch) to tensorflow, but I think this feature would generally benefit many projects that are related to the growing field of meta-learning.

**Any Other info.**
Thanks and Merry Christmas!!!

**[Contributing](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md)**

- Do you want to contribute a PR? (yes/no): no
- If yes, please read [this page](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md) for instructions
- Briefly describe your candidate solution(if contributing):

**Comment by tilakrayal** [comment]
@NuoWenLei ,
Can you please share a reproducible code that supports your statement so that the issue can be easily understood? Thanks!

**Comment by NuoWenLei** [comment]
Yes, pleas refer to 
[this colab notebook](https://colab.research.google.com/drive/1JU6n3N1wX7NBsa3i_PqKR0mSUCYmYKO3?usp=sharing)

**Comment by chenmoneygithub** [comment]
set_weights is not expected to be differentiable, thanks for the issue!
